o11y/otel/otel.go

```diff
@@ -208,7 +208,7 @@ type span struct {
 	metricsProvider o11y.ClosableMetricsProvider
 	start           time.Time
 
-	mu     sync.Mutex // mu is a write mutex for the map below (concurrent reads are safe)
+	mu     sync.RWMutex // mu is a write mutex for the map below (concurrent reads are safe)
```

Comment:
are concurrent reads safe? since we're now read locking, perhaps that part of the comment should be removed

Reply:
That's 100% safe - many places use an unmodified map as a lookup between go-routines.